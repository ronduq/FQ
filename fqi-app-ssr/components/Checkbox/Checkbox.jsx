import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import s from './Checkbox.module.scss';

class Checkbox extends React.Component {

  handleCheckboxChange = event => {
    const {
      updateItem,  
      index, 
      type, 
      selectAllItems, 
      parentCode, 
      isParent,
      currentRetailer,
      criteria
    } = this.props;

    this.setState({ checked: event.target.checked });
    if (type === 'all') {
      selectAllItems();
    } else {
      if (currentRetailer){
        updateItem(index, event.target.checked, parentCode, isParent,'',currentRetailer);
      } else {
        updateItem(index, event.target.checked, parentCode, isParent, criteria);
      }
      
    }
  }

  render() {
    const {
      label,
      type,
      selected,
      allNumber,
      name,
      disabled,
    } = this.props;

    return (
      <div className={classNames(s.checkbox,{ [s.checkboxChild]: type === 'child', [s.checkboxParent]: type === 'parent'})}>
        <input 
          type="checkbox" 
          id={`${label}${name}`} 
          checked={selected}
          onChange={this.handleCheckboxChange}
          disabled={disabled}
          autoComplete="off"
        />
        <label htmlFor={`${label}${name}`} aria-hidden="true"><span>{label}{ type === 'all' ? ` (${allNumber})` : ''}</span></label>
      </div>
    );
  }
};

Checkbox.propTypes = {
  label: PropTypes.string,
  checked: PropTypes.bool,
  type: PropTypes.string,
};

export default Checkbox;