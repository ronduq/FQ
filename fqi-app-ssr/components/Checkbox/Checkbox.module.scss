@import '../../style/settings/colours';
@import '../../style/settings/fonts';

.checkbox input[type="checkbox"] {
  opacity: 0;
  position: absolute;
}

.checkbox label {
  position: relative;
  display: inline-block;
  padding-left: 34px;
  padding-bottom: 2px;
  @include font-family('proxima-regular');
  @include font-size('medium');
  width: 100%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  cursor: pointer;
  box-sizing: border-box;
  
  &:hover {
    text-decoration: underline;
  }
}

.checkbox label::before,
.checkbox label::after {
  position: absolute;
  content: "";
  display: inline-block;
  transition: ease-in-out .1s all;
}

/*Outer box of the fake checkbox*/
.checkbox input[type="checkbox"] + label::before{
  height: 24px;	
  width: 24px;	
  border-radius: 2px;	
  background-color: $colour-secondary;
  left: 0;
  top: 0;
}
.checkboxChild input[type="checkbox"] + label::before{
  background-color: $colour-secondary-light;
}

.checkbox input[type="checkbox"]:focus  + label::before{
  box-shadow: 0 0 0 2px dodgerblue
}

.checkbox input[type="checkbox"]:not(:checked) + label::before{
  box-sizing: border-box;
  border: 1px solid $colour-grey-medium;
  background-color: $colour-grey-light;
}

.checkbox input[type="checkbox"]:disabled + label {
  &:hover {
    text-decoration: none;
  }

  span { 
    cursor: default;
    opacity: 0.5;
  }
}

/*Checkmark of the fake checkbox*/
.checkbox label::after {
  color: white;
  height: 7px;
  width: 14px;
  border-left: 2px solid;
  border-bottom: 2px solid;
  transform: rotate(-45deg);
  left: 4px;
  top: 4px;
}

/* Line of the fake secondary checkbox*/

.checkboxParent {
  display: block;
}

/*Hide the checkmark by default*/
.checkbox input[type="checkbox"] + label::after {
  content: none;
}

/*Unhide on the checked state*/
.checkbox input[type="checkbox"]:checked + label::after {
  content: "";
}

/*Adding focus styles on the outer-box of the fake checkbox*/
.checkbox input[type="checkbox"]:focus + label::before {
  outline: rgb(59, 153, 252) auto 5px;
}