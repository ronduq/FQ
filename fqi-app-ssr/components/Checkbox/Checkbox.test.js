/* eslint-env jest */

import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Checkbox from './Checkbox'


describe('With Snapshot Testing', () => {
  const handleCheckboxChange = jest.fn();
  const updateItem = jest.fn();
  const selectAllItemsCheckbox = jest.fn();
  const selectAllItems = jest.fn();

  const props = {
    index: 0,
    isParent: false,
    label: "Costco",
    selected: false,
    type: "",
    updateItem: updateItem,
    selectAllItems: selectAllItems
  }
  

  it('Checkbox snapshot"', () => {
    const component = renderer.create(<Checkbox />)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  })

  it('should fire handleChecupdateItemboxChange', () => {
    const wrapper = mount(<Checkbox {...props} />);
    const event = {target: {checked: true}}
    wrapper.find('input').simulate('change', event);
    expect(updateItem).toBeCalledTimes(1);
  })

  it('should fire handleCheckboxChange selectall', () => {
    const wrapper = mount(<Checkbox {...props} type='all'/>);
    const event = {target: {checked: true}}
    wrapper.find('input').simulate('change', event);
    expect(selectAllItems).toBeCalledTimes(1);
  })

  it('should set disabled attr to input when passed in props', () => {
    const wrapper = mount(<Checkbox {...props} disabled />);
    expect(wrapper.find('input[disabled]').length).toBe(1);
  });
})