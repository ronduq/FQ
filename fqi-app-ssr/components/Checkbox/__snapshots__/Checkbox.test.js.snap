// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Checkbox snapshot" 1`] = `
<div
  className="checkbox"
>
  <input
    autoComplete="off"
    id="undefinedundefined"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    aria-hidden="true"
    htmlFor="undefinedundefined"
  >
    <span>
      
    </span>
  </label>
</div>
`;
